### PR TITLE
Add blog post of Codemotion Madrid

### PR DESCRIPTION
**Pull request by jangelgarcia**

### CODE REVIEW

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: "Codemotion 2015 - One ePagees was there!"
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: Codemotion 2015 - One ePagees was there! > Codemotion 2015 - One ePagee was there!

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+date: "2015-12-07 12:00:00"
+icon: search
+categories: conferences
+authors: ["Ángel García"]
+---
+The Codemotion conference took place in Madrid on November 28 and 29. ePages was present there. With more than 1900 software enthusiasts, Codemotion is the biggest IT event in Spain. This conference was sponsored by companies such as IBM, Microsoft, Arsys, Mozilla, Google... Two days fulled of talks from 12 different tracks, where each track
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: ePages was present there. > I would leave that out as it is mentioned in the title already.

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: "Codemotion 2015 - One ePagees was there!"
+date: "2015-12-07 12:00:00"
+icon: search
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: maybe use the comments icon?

File: _posts/2015/2015-12-07-codemotion-madrid.md (hunk 2 above)
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: Two days fulled of talks... > Two days filled with talks... or Two days full of talks...

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+focussed on a specific theme such as  Software Quality, Agile Methodologies, Clean Code, Testing, Big Data, Infrastructure,
+Scala, Java, Cassandra, MongoDB, Apache Drill, Selenium, Spark... Some of them have been broadcasted, and now you have the chance to watch them on [Youtube](http://2015.codemotion.es/streaming.html) (unfortunately only available in Spanish).
+
+It was a good opportunity to get inspiration and I really enjoyed the variety of talks, which allowed me to get to know new technologies and discover different points of view on how to deal with tricky situations in some of the most well-known technology companies. From my point of view, each talk had different levels of difficuly and quality, I think that most of the speakers were quite experts in the topic. I would like to highlight here some talks that I consider the most interesting.
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: From my point of view, each talk had different levels of difficuly and quality, > Make a fullstop behind quality.
Start with a new sentence: I think that most of the ...

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+#The process for having quality software in 4 steps
+This talk was about how to deal with the challenge of designing software with guarantees for a quality result, and the methodology used by IBM to achieve that goal. Basically, the used methodology is called "Design Thinking", and it keeps the user at the center from the very beginning until the final phases of development, taking care of user experience and using technologies and frameworks such as Angular, jQuery, HTML5 and CSS3. It’s based on four core principles: Understand, Explore, Prototype, Evaluate. The main goal of "Design Thinking" is to create a real and intuitive experience, understanding the users’ needs, weak points, exploring solutions, prototyping the experience and evaluating results.
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: Amend this sentence: Basically, the used methodology is called "Design Thinking". It keeps the user at the centre from the very beginning until the final phases of development, taking care of user experience and using technologies and frameworks such as Angular, jQuery, HTML5 and CSS3.

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+#Love your tests and Microservices testing
+Some companies are migrating their huge monolithic applications towards something called service-oriented. One of these companies is [Tuenti](http://www.tuenti.com/), which had to face plenty of challenges to migrate its social network code in order to become a high-tech virtual mobile network operator integrating different third party services. It is said that code tests take before deliverying the product to production envs, but unfortunately most of technology companies have several issues in real environments. Some given recommendations in this talk were:
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: It is said that code tests take before deliverying the product to production envs,... > It is said that code tests take before delivering the product to production environments,...

Question: is there a word missing in this phrase: It is said that code tests take before delivering

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+* Make different tests: unit, integration (services and API) and UI. It is recommended to create a 70% of Unit tests, a 20% of Integration test and 10% of Acceptance test.
+* Avoid Ice-cream cone anti-pattern.
+* Refactoring old code, and more refactoring after refactoring...
+* Clean code: describes a lot of principles and "best practices".
+* Mocks, mocks everywhere.
+* Use Name conventions (human readable).
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: Use naming conventions

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+* Given-When-Then is a style of representing tests, specifying a system's behavior.
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: behavior > behaviour

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+* Software development needs professional people, making test is professional!
```
Comment (BirgitBader) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: making test is professional > making tests is professional

**Comment by BirgitBader** [comment]
@maletzke @steffimeier please give your feedback on this post, thanks.

**Comment by BirgitBader** [comment]
@JessTinker please proof-read, thanks.

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: "Codemotion 2015 - One ePagee was there!"
+date: "2015-12-07 12:00:00"
+icon: search
+categories: comments
+authors: ["Ángel García"]
+---
+The Codemotion conference took place in Madrid on November 28 and 29. With more than 1900 software enthusiasts, Codemotion is the biggest IT event in Spain. This conference was sponsored by companies such as IBM, Microsoft, Arsys, Mozilla, Google... Two days filled with talks from 12 different tracks, where each track
```
Comment (JessTinker) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: This conference was sponsored by companies such as IBM, Microsoft, Arsys, Mozilla and Google... Two days filled with talks from 12 different tracks, where each track

File: _posts/2015/2015-12-07-codemotion-madrid.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+focussed on a specific theme such as  Software Quality, Agile Methodologies, Clean Code, Testing, Big Data, Infrastructure,
+Scala, Java, Cassandra, MongoDB, Apache Drill, Selenium, Spark... Some of them have been broadcasted, and now you have the chance to watch them on [Youtube](http://2015.codemotion.es/streaming.html) (unfortunately only available in Spanish).
```
Comment (JessTinker) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: +Scala, Java, Cassandra, MongoDB, Apache Drill, Selenium and Spark...

Comment (JessTinker) [inline _posts/2015/2015-12-07-codemotion-madrid.md]: Some of them have been broadcasted, giving you the chance to watch them on [Youtube](http://2015.codemotion.es/streaming.html) (unfortunately only available in Spanish).

**Pull request merged**